Bottom: 6c4d00650d47cbb6cfaadffe733bf35821e043c3
Top:    a29724a3259a572382785eac942f74aac856b61a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 15:13:45 +0000

Fixup move of step stuff


---

diff --git a/src/Makefile.am b/src/Makefile.am
index df42091..bd6874d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -187,9 +187,7 @@ LIBGTK_GDK_SRCS= \
 LIBGTK_GL_SRCS= \
 	hid/gtk/gtkhid-gl.c \
 	hid/gtk/gui-trackball.c \
-	hid/gtk/gui-trackball.h \
-	hid/gtk/step.c \
-	hid/gtk/step.h
+	hid/gtk/gui-trackball.h
 
 GL_SRCS= \
 	hid/common/hidgl.c \
@@ -361,7 +359,11 @@ LIBGTK_SRCS = \
 	hid/gtk/gui-top-window.c \
 	hid/gtk/gui-utils.c \
 	hid/gtk/snavi.c \
-	hid/gtk/snavi.h
+	hid/gtk/snavi.h \
+	hid/step/step_writer.c \
+	hid/step/step_writer.h \
+	hid/step/object3d_step.c \
+	hid/step/object3d_step.h
 libgtk_a_SOURCES = ${LIBGTK_SRCS} hid/gtk/gtk_lists.h
 
 hid/gtk/gtk_lists.h : ${LIBGTK_SRCS} Makefile
diff --git a/src/hid/common/vertex3d.c b/src/hid/common/vertex3d.c
index 8abef8f..ff3ab98 100644
--- a/src/hid/common/vertex3d.c
+++ b/src/hid/common/vertex3d.c
@@ -3,7 +3,7 @@
 #include <stdbool.h>
 #include <stdio.h>
 
-//#include "step.h"
+#include "step_id.h"
 #include "vertex3d.h"
 
 static int global_vertex3d_count;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 491154b..9b91a07 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -15,7 +15,7 @@
 #include "gui-pinout-preview.h"
 #include "pcb-printf.h"
 
-#include "step.h"
+#include "hid/step/step_writer.h"
 #include "hid/common/quad.h"
 #include "hid/common/vertex3d.h"
 #include "hid/common/contour3d.h"
diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
new file mode 100644
index 0000000..0d4db9d
--- /dev/null
+++ b/src/hid/step/object3d_step.c
@@ -0,0 +1,24 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <stdbool.h>
+
+#include <glib.h>
+
+#include "hid/common/step_id.h"
+#include "hid/common/quad.h"
+#include "hid/common/vertex3d.h"
+#include "hid/common/contour3d.h"
+#include "hid/common/appearance.h"
+#include "hid/common/face3d.h"
+#include "hid/common/edge3d.h"
+#include "hid/common/object3d.h"
+#include "polygon.h"
+#include "data.h"
+
+#include "step_writer.h"
+
+#include "pcb-printf.h"
+
+#include "object3d_step.h"
+
diff --git a/src/hid/step/object3d_step.h b/src/hid/step/object3d_step.h
new file mode 100644
index 0000000..5dc6f4d
--- /dev/null
+++ b/src/hid/step/object3d_step.h
@@ -0,0 +1,2 @@
+void object3d_list_export_to_step_assy (GList *objects, const char *filename);
+void object3d_export_to_step (object3d *object, const char *filename);
diff --git a/src/hid/gtk/step.c b/src/hid/step/step_writer.c
similarity index 99%
rename from src/hid/gtk/step.c
rename to src/hid/step/step_writer.c
index 090a338..0a04ea0 100644
--- a/src/hid/gtk/step.c
+++ b/src/hid/step/step_writer.c
@@ -3,7 +3,7 @@
 #include <stdio.h>
 #include <stdarg.h>
 
-#include "step.h"
+#include "step_writer.h"
 
 static char *
 step_bool (bool expr)
diff --git a/src/hid/gtk/step.h b/src/hid/step/step_writer.h
similarity index 100%
rename from src/hid/gtk/step.h
rename to src/hid/step/step_writer.h
